<?xml version="1.0"?>
<launch>
  <arg name="ns"              default="/"      doc="Namespace this robot is in" />
  <arg name="rviz"            default="true"   doc="Should rviz be spawned?" />
  <arg name="rvizfile"        default="$(find panda_push_mujoco)/launch/rviz/panda_pokingStick_with_marker.rviz" />
  <arg name="arm_id"          default="panda"  doc="Name of the panda robot to spawn" />
  <arg name="controller"      default="joint_velocity_push_controller" doc="Which example controller should be started?" />

  <include file="$(find panda_push_sim2real)/launch/panda_push.launch" pass_all_args="true" />

  <!-- joint_velocity_push_controller -->
  <node name="interactive_marker_movement" 
        pkg="panda_push_sim2real" 
        type="interactive_marker.py" 
        required="true" 
        output="screen"
        if="$(eval arg('controller') == 'joint_velocity_push_controller')">
    <param name="link_name" value="$(arg arm_id)_link0" />
    <remap from="equilibrium_pose" to="/joint_velocity_push_controller/equilibrium_pose" />
  </node>

  <!-- cartesian_impedance_example_controller -->
  <node name="interactive_marker_movement" 
        pkg="sim2real" 
        type="interactive_marker.py" 
        required="true" 
        output="screen"
        if="$(eval arg('controller') == 'cartesian_impedance_example_controller')">
    <param name="link_name" value="$(arg arm_id)_link0" />
    <remap from="equilibrium_pose" to="/cartesian_impedance_example_controller/equilibrium_pose" />
  </node>

</launch>